dev/devicelab/lib/tasks/plugin_tests.dart

```diff
@@ -494,42 +518,55 @@ s.dependency 'AppAuth', '1.6.0'
         }
 
         if (validateNativeBuildProject) {
-          final File podsProject = File(path.join(rootPath, target, 'Pods', 'Pods.xcodeproj', 'project.pbxproj'));
-          if (!podsProject.existsSync()) {
-            throw TaskResult.failure('Xcode Pods project file missing at ${podsProject.path}');
-          }
-
-          final String podsProjectContent = podsProject.readAsStringSync();
-          if (target == 'ios') {
-            // Plugins with versions lower than the app version should not have IPHONEOS_DEPLOYMENT_TARGET set.
-            // The plugintest plugin target should not have IPHONEOS_DEPLOYMENT_TARGET set since it has been lowered
-            // in _reduceDarwinPluginMinimumVersion to 10, which is below the target version of 11.
-            if (podsProjectContent.contains('IPHONEOS_DEPLOYMENT_TARGET = 10')) {
-              throw TaskResult.failure('Plugin build setting IPHONEOS_DEPLOYMENT_TARGET not removed');
-            }
-            // Transitive dependency AppAuth targeting too-low 8.0 was not fixed.
-            if (podsProjectContent.contains('IPHONEOS_DEPLOYMENT_TARGET = 8')) {
-              throw TaskResult.failure('Transitive dependency build setting IPHONEOS_DEPLOYMENT_TARGET=8 not removed');
+          final File generatedSwiftManifest = File(path.join(
+            rootPath,
+            target,
+            'Flutter',
+            'ephemeral',
+            'Packages',
+            'FlutterGeneratedPluginSwiftPackage',
+            'Package.swift'
+          ));
+          final bool swiftPackageManagerEnabled = generatedSwiftManifest.existsSync();
+
+          if (!swiftPackageManagerEnabled) {
```

Comment:
I recommend reviewing this diff with whitespace disabled: https://github.com/flutter/flutter/pull/152049/files?diff=split&w=1
 
![image](https://github.com/user-attachments/assets/f799fb58-7f6b-40e7-b54d-0e28adcffdca)